FreeSurfer/LongitudinalFreeSurferPipeline.sh

```diff
@@ -0,0 +1,368 @@
+
+  if [[ $(( ${freesurfer_primary_version} )) -lt 6 ]]; then
+    # e.g. 4.y.z, 5.y.z
+    log_Err_Abort "FreeSurfer version 6.0.0 or greater is required. (Use FreeSurferPipeline-v5.3.0-HCP.sh if you want to continue using FreeSurfer 5.3)"
+  fi
+}
+
+# Configure the use of FreeSurfer v6 custom tools
+configure_custom_tools
+
+# Show tool versions
+show_tool_versions
+
+# Validate version of FreeSurfer in use
+validate_freesurfer_version
+
+#processing code goes here
+echo "parallel mode: $parallel_mode"
+fslsub_queue=NONE
+
+case $parallel_mode in
+	FSLSUB)
+    if [ -n "$queue" ]; then fslsub_queue=$queue; fi	
+    ;;    
+	BUILTIN)
+		max_jobs=$max_jobs
+    if (( max_jobs < 1 )); then max_jobs=2; fi 
+    ;;
+	NONE)
+    echo "parallel mode: NONE" 
+    ;;
+	*)
+		log_Err_Abort "Unknown parallel mode $parallel_mode. Plese specify one of FSLSUB, BUILTIN, NONE"
+		;;
+esac
+
+start_stage=0
+if [ -n "$StartStage" ]; then	
+	case $StartStage in
+    TEMPLATE) start_stage=0 ;;
+		TIMEPOINTS) start_stage=1 ;;
+		*) log_Err_Abort "Unrecognized option for start-stage: $StartStage"
+	esac
+fi
+end_stage=1
+if [ -n "$EndStage" ]; then	
+	case $EndStage in
+		TEMPLATE) end_stage=0 ;;
+		TIMEPOINTS) end_stage=1 ;;
+		*) log_Err_Abort "Unrecognized option for end-stage: $EndStage"
+	esac
+fi
+
+# ----------------------------------------------------------------------
+log_Msg "Starting main functionality"
+# ----------------------------------------------------------------------
+
+# ----------------------------------------------------------------------
+# Log values retrieved from positional parameters
+# ----------------------------------------------------------------------
+log_Msg "StudyFolder: ${StudyFolder}"
+log_Msg "SubjectID: ${SubjectID}"
+log_Msg "Sessions: ${Sessions}"
+log_Msg "TemplateID: ${TemplateID}"
+log_Msg "recon_all_seed: ${recon_all_seed}"
+log_Msg "StartStage: $StartStage"
+log_Msg "EndStage: $EndStage"
+
+# ----------------------------------------------------------------------
+log_Msg "Preparing the folder structure"
+# ----------------------------------------------------------------------
+Sessions=`echo ${Sessions} | sed 's/,/ /g'`
+
+extra_reconall_args_base=""
+extra_reconall_args_long=""
+
+Session0=( $Sessions ); Session0=${Session0[0]}
+if (( UseT2w )); then
+  extra_reconall_args_base="-T2pial -T2 ${StudyFolder}/${Session0}/T1w/T2w_acpc_dc_restore.nii.gz"
+  extra_reconall_args_long="-T2pial"
+fi
+
+log_Msg "extra_reconall_args_base: $extra_reconall_args_base"
+log_Msg "extra_reconall_args_long: $extra_reconall_args_long"
+log_Msg "After delimiter substitution, Sessions: ${Sessions}"
+
+LongDIR="${StudyFolder}/${SubjectID}.long.${TemplateID}/T1w"
+mkdir -p "${LongDIR}"
+
+for Session in ${Sessions} ; do
+  Source="${StudyFolder}/${Session}/T1w/${Session}"
+  Target="${LongDIR}/${Session}"
+  log_Msg "Creating a link: ${Source} => ${Target}"
+  ln -sf ${Source} ${Target}
+done
```

Comment:
Again, should this be run regardless of the setting of start and end stages?

For that matter, why do we need to support "start and end stage" in this script?

Reply:
1 Agree. This only needs to run once at the TEMPLATE stage.
2 To support staged parallel execution, where next stage depends on previous one. With only two stages one might use 'process template' and 'process timepoints' bool flags, but staging is more consistent with subsequent processing.

Reply:
What benefit to the user is there in being able to run only one of the stages?  Users generally would prefer to just push one button, wait, and get the final results, so intentionally making things more complicated needs to come with some benefit.